lib/Dialect/AIEVec/Transforms/ConvertVectorToAIEVec.cpp

```diff
@@ -0,0 +1,244 @@
+}
+
+static void populateAIEVecV1ConversionPatterns(RewritePatternSet &patterns,
+                                               AnalysisManager &am) {
+  patterns.add<SplitUPDOpOnAccPattern>(patterns.getContext(), am, 256);
+}
+
+static void populateAIEVecV2ConversionPatterns(RewritePatternSet &patterns,
+                                               AnalysisManager &am) {
+  patterns.add<SplitUPDOpOnAccPattern>(patterns.getContext(), am, 512);
+}
+
+//===----------------------------------------------------------------------===//
+// Legalizations
+//===----------------------------------------------------------------------===//
+
+static void configureAIEVecCommonLegalizations(ConversionTarget &target,
+                                               AnalysisManager &am) {
+  target.addLegalDialect<xilinx::aievec::AIEVecDialect>();
+  target.addIllegalOp<vector::TransferReadOp>();
+  target.addDynamicallyLegalOp<arith::AddIOp>(
+      [](arith::AddIOp op) { return !isa<VectorType>(op.getType()); });
+  target.addDynamicallyLegalOp<arith::AddFOp>(
+      [](arith::AddFOp op) { return !isa<VectorType>(op.getType()); });
+  target.addDynamicallyLegalOp<arith::SubIOp>(
+      [](arith::SubIOp op) { return !isa<VectorType>(op.getType()); });
+  target.addDynamicallyLegalOp<arith::SubFOp>(
+      [](arith::SubFOp op) { return !isa<VectorType>(op.getType()); });
+}
+
+static void configureAIEVecV1Legalizations(ConversionTarget &target,
+                                           AnalysisManager &am) {
+  target.addDynamicallyLegalOp<aievec::UPDOp>([&am](xilinx::aievec::UPDOp op) {
+    return am.getChildAnalysis<UPDOpEffectiveAccessSizeAnalysis>(op)
+               .effectiveSize <= 512;
+  });
+}
+
+static void configureAIEVecV2Legalizations(ConversionTarget &target,
+                                           AnalysisManager &am) {
+  target.addDynamicallyLegalOp<aievec::UPDOp>([&am](aievec::UPDOp op) {
+    return am.getChildAnalysis<UPDOpEffectiveAccessSizeAnalysis>(op)
+               .effectiveSize <= 1024;
+  });
+}
+
+//===----------------------------------------------------------------------===//
+// Conversion pass
+//===----------------------------------------------------------------------===//
+struct ConvertVectorToAIEVec
+    : public aievec::impl::ConvertVectorToAIEVecBase<ConvertVectorToAIEVec> {
+  using Base::Base;
+
+  void runOnOperation() override;
+};
+
+std::unique_ptr<Pass> xilinx::aievec::createConvertVectorToAIEVecPass(
+    const ConvertVectorToAIEVecOptions &options) {
+  return std::make_unique<ConvertVectorToAIEVec>(options);
+}
+
+/// Convert incoming vector operations into their corresponding AIE vector
+/// intrinsics. The pass assumes the IR is in a state
+void ConvertVectorToAIEVec::runOnOperation() {
+  auto func = getOperation();
+  MLIRContext *context = &getContext();
+  RewritePatternSet patterns(context);
+  ConversionTarget target(*context);
+  int8_t aieVersion = 1;
```

Comment:
It's probably worthwhile defining an enum type for specifying the AIE architecture. e.g.,
enum class AieArch {
 aie,
 aie_ml
};

Reply:
Yes, I was thinking the same.